src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1356,25 +1372,23 @@ private void PortsCollectionChanged(object sender, System.Collections.Specialize
                     ConfigureSnapEdges(sender == InPorts ? InPorts : OutPorts);
                     foreach (PortModel p in e.NewItems)
                     {
+                        p.PropertyChanged += OnPortPropertyChanged;
+
                         p.Connectors.CollectionChanged += (coll, args) =>
                         {
                             // Call the collection changed handler, replacing
                             // the 'sender' with the port, which is required
                             // for the disconnect operations.
                             ConnectorsCollectionChanged(p, args);
                         };
```

Comment:
What about unsubscribing `p.Connectors.CollectionChanged`? Since it's attached to an anonymous function, you can't unsubscribe it unless you name the function (lambda).